File: backend/src/main/java/com/api/bompreparo/data/repositories/ImageRepository.java
package com.api.bompreparo.data.repositories;

import com.api.bompreparo.domain.models.Image;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface ImageRepository extends JpaRepository<Image, Long> {
}

File: frontend/bom-preparo/src/app/data/repositories/image.repository.ts
import { HttpClient } from "@angular/common/http";
import { Injectable } from "@angular/core";
import { Image } from "src/app/domain/models";

import endpoints from "../../core/sources/api.source"
import { environment } from "src/environments/environment";
import { Observable, map } from "rxjs";

@Injectable({ providedIn: 'root' })
export class ImageRepository {

  constructor(private httpClient: HttpClient) { }

  viewImage(imageId: number): Observable<any> {
    return this.httpClient.get<any>(environment.apiUrl + endpoints.image.viewImage + "?imageId=" + imageId).pipe(
      map((data) => {
        return data;
      })
    );
  }

}
